sdk/azidentity/default_azure_credential.go

```diff
@@ -87,3 +90,20 @@ func NewDefaultAzureCredential(options *DefaultAzureCredentialOptions) (*Default
 func (c *DefaultAzureCredential) GetToken(ctx context.Context, opts policy.TokenRequestOptions) (token *azcore.AccessToken, err error) {
 	return c.chain.GetToken(ctx, opts)
 }
+
+func defaultAzureCredentialErrorHandler(numberOfSuccessfulCredentials int, errorMessages []string) (err error) {
+	errorMessage := "Default Azure Credential:\n\t" + strings.Join(errorMessages, "\n\t")
+
+	if numberOfSuccessfulCredentials == 0 {
+		err := fmt.Errorf("Authentication failed for the %s", errorMessage)
+		authErr := newAuthenticationFailedError(err, nil)
```

Comment:
Only the ctor calls this function. The credential hasn't attempted to authenticate

Reply:
Updated here: https://github.com/Azure/azure-sdk-for-go/pull/16844/commits/307bd457f6e6b6e7e5696c6382e2bfbe211a68d0